### PR TITLE
[Bug-fix] Fix loose hanging onFailure Nodes of email/sms send executors

### LINKED ISSUE #3980: Templating Failure after a Input Validation Fails on First try and the user retries
### Description

If a user tries to follow such as SMS OTP, and the input validation fails, on the first time, and then properly enters the correct phone number, in the SMS sent the templating is not properly applied.

### Steps to Reproduce

1. SetUp thunderId
2. SetUp a SMS sender
3. SetUp a SignUp flow, which uses the SMS sender such as SMS OTP
4. SetUp a client application, such as a React Application
5. Select the previously (Step 3) flow to the application
6. Try to SignUp, when asked for a phone number, enter an invalid number, example, 1111

Image

Image

7. When prompted to enter the correct number the second time, enter a valid number, such as 1111111111, so the flow requests for a valid code.

Image

8. Check the SMS or webhook, received.

Image

### Version

0.48

### Environment Details (with versions)

MacOS with webhook for Custom SMS

### PR DESCRIPTION
**Pull request by NutharaNR**
### Purpose
Fixes: https://github.com/thunder-id/thunderid/issues/3980

### Related Issues
- https://github.com/thunder-id/thunderid/issues/3980

### Related PRs
- N/A

### Checklist
- [ ] Followed the contribution guidelines.
- [ ] Manual test round performed and verified.
- [ ] Documentation provided. (Add links if there are any)
    - [ ] Ran Vale and fixed all errors and warnings
- [ ] Tests provided. (Add links if there are any)
    - [ ] Unit Tests
    - [ ] Integration Tests
- [ ] Breaking changes. (Fill if applicable)
    - [ ] Breaking changes section filled.
    - [ ] `breaking change` label added.

### Security checks
- [ ] Followed secure coding standards in [WSO2 Secure Coding Guidelines](https://security.docs.wso2.com/en/latest/security-guidelines/secure-engineering-guidelines/secure-coding-guidlines/introduction/)
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved SMS and email OTP flows by routing OTP-send failures back to the correct phone/email entry prompts.
  * Updated additional registration flows so OTP-send failures return users to the appropriate mobile entry step.
  * Refined flow transition handling so failure/incomplete outcomes follow configured `onFailure`/`onIncomplete`, including “failure without error” cases.

* **Tests**
  * Updated SMS OTP recovery expectations for invalid OTP submissions and aligned flow configuration for `verify_otp` incomplete behavior.
  * Expanded unit and integration coverage for failure/incomplete edge cases and input-clearing behavior.

### CODE REVIEW

File: tests/integration/flow/recovery/sms_otp_recovery_test.go [hunk 1]
```diff
@@ -491,8 +491,9 @@ func buildSMSOTPRecoveryFlow(senderID string) testutils.Flow {
 					"name": "OTPExecutor",
 					"mode": "verify",
 				},
-				"onSuccess": "prompt_new_password",
-				"onFailure": "prompt_username",
+				"onSuccess":    "prompt_new_password",
+				"onFailure":    "prompt_username",
+				"onIncomplete": "otp_sent_status",
```
[the remark this answers is not in the document]
Comment (NutharaNR) [reply tests/integration/flow/recovery/sms_otp_recovery_test.go:496]: not needed to document. this is an integration test with a custom test scenario

**Review by ThaminduDilshan — APPROVED** [review]
